packages/paymasters/contracts/PermitERC20UniswapV3Paymaster.sol

```diff
@@ -15,71 +16,79 @@ import "@opengsn/contracts/src/utils/GsnUtils.sol";
 import "./interfaces/IChainlinkOracle.sol";
 
 import "./helpers/UniswapV3Helper.sol";
-import "./helpers/PermitInterfaceDAI.sol";
+import "../../contracts/src/interfaces/IERC20Token.sol";
 
 /**
  * A paymaster allowing addresses holding ERC20 tokens with 'permit' functionality
  * to pay for a GSN transaction.
  */
 contract PermitERC20UniswapV3Paymaster is BasePaymaster, ERC2771Recipient {
 
+    using SafeERC20 for IERC20Metadata;
+
     event Received(address indexed sender, uint256 eth);
-    event TokensCharged(uint256 gasUseWithoutPost, uint256 gasJustPost, uint256 ethActualCharge);
+    event TokensCharged(uint256 gasUseWithoutPost, uint256 gasJustPost, uint256 tokenActualCharge, uint256 ethActualCharge);
 
-    IChainlinkOracle public priceFeed;
-    ISwapRouter public uniswap;
-    IERC20 public token;
-    IERC20 public weth;
+    struct TokenSwapData {
+        IChainlinkOracle priceFeed;
+        uint24 uniswapPoolFee;
+        // between 0 to 1000, with 2 decimals, that is, 10 = 1%
+        uint8 slippage;
```

Comment:
what is the meaning of slippage ? 
it is the name and type of uniswap param, but you don't use it for that purpose at all.

Reply:
It's not the name of a uniswap param